crates/bevy_core/src/time/timer.rs

```diff
@@ -364,6 +367,334 @@ impl Timer {
+    /// let mut repeating = DiscreteTimer::new(2, true);
+    /// timer.tick(3);
+    /// repeating.tick(3);
+    /// assert_eq!(timer.elapsed(), 2);
+    /// assert_eq!(repeating.elapsed(), 1);
+    /// ```
+    pub fn tick(&mut self, delta: u64) -> &Self {
+        if self.paused() {
+            return self;
+        }
+
+        if !self.repeating() && self.finished() {
+            self.times_finished = 0;
+            return self;
+        }
+
+        self.stopwatch.tick(delta);
+        self.finished = self.elapsed() >= self.duration();
+
+        if self.finished() {
+            if self.repeating() {
+                self.times_finished = (self.elapsed() / self.duration()) as u32;
+                self.set_elapsed(self.elapsed() % self.duration());
+            } else {
+                self.times_finished = 1;
+                self.set_elapsed(self.duration());
+            }
+        } else {
+            self.times_finished = 0;
+        }
+
+        self
+    }
+
+    /// Pauses the Timer. Disables the ticking of the timer.
+    ///
+    /// See also [`Stopwatch::pause`](Stopwatch::pause).
+    ///
+    /// # Examples
+    /// ```
+    /// # use bevy_core::*;
+    /// let mut timer = DiscreteTimer::new(2, false);
+    /// timer.pause();
+    /// timer.tick(1);
+    /// assert_eq!(timer.elapsed(), 0);
+    /// ```
+    #[inline]
+    pub fn pause(&mut self) {
+        self.stopwatch.pause();
+    }
+
+    /// Unpauses the Timer. Resumes the ticking of the timer.
+    ///
+    /// See also [`Stopwatch::unpause()`](Stopwatch::unpause).
+    ///
+    /// # Examples
+    /// ```
+    /// # use bevy_core::*;
+    /// let mut timer = DiscreteTimer::new(3, false);
+    /// timer.pause();
+    /// timer.tick(1);
+    /// timer.unpause();
+    /// timer.tick(2);
+    /// assert_eq!(timer.elapsed(), 2);
+    /// ```
+    #[inline]
+    pub fn unpause(&mut self) {
+        self.stopwatch.unpause();
+    }
+
+    /// Returns `true` if the timer is paused.
+    ///
+    /// See also [`Stopwatch::paused`](Stopwatch::paused).
+    ///
+    /// # Examples
+    /// ```
+    /// # use bevy_core::*;
+    /// let mut timer = DiscreteTimer::new(1, false);;
+    /// assert!(!timer.paused());
+    /// timer.pause();
+    /// assert!(timer.paused());
+    /// timer.unpause();
+    /// assert!(!timer.paused());
+    /// ```
+    #[inline]
+    pub fn paused(&self) -> bool {
+        self.stopwatch.paused()
+    }
+
+    /// Resets the timer. the reset doesn't affect the `paused` state of the timer.
+    ///
+    /// See also [`Stopwatch::reset`](Stopwatch::reset).
+    ///
+    /// Examples
+    /// ```
+    /// # use bevy_core::*;
+    /// use std::time::Duration;
+    /// let mut timer = Timer::from_seconds(1.0, false);
+    /// timer.tick(Duration::from_secs_f32(1.5));
+    /// timer.reset();
+    /// assert!(!timer.finished());
+    /// assert!(!timer.just_finished());
+    /// assert_eq!(timer.elapsed_secs(), 0.0);
+    /// ```
```

Comment:
should be
```rust
    /// ```
    /// # use bevy_core::*;
    /// let mut timer = DiscreteTimer::new(1, false);
    /// timer.tick(2);
    /// timer.reset();
    /// assert!(!timer.finished());
    /// assert!(!timer.just_finished());
    /// assert_eq!(timer.elapsed(), 0);
    /// ```
```